### DISCUSSION TITLE
¿Cómo validar un form que usa Wallet?

### DISCUSSION THREAD

**Discussion by acyment** [mercadopago/sdk-react#96] [Q&A] [answered]
Estoy usando el componente Wallet como botón de un form y quisiera poder implementar alguna de estas dos estrategias para la validación del form:
- Cambiar el Wallet a un estado "disabled" (como se haría con un botón típico de Submit - idealmente gris y con cursor de not-allowed). No encuentro ningun atributo que permita hacer esto.
- Capturar el onSubmit y poder frenar el proceso si el form no valida. Cuando hago esto (por ejemplo devolviendo un Promise con reject) el form no hace el submit, pero aparece el cartel de Redirigiendo y el botón no vuelve a aparecer

Entiendo que es un caso de uso bastante común, pero no logro encontrar un ejemplo que lo implemente de estas u otras maneras. Muchas gracias

**Comment by icaldana** [comment]
Hola, buenas!

En primer lugar, perdón por la demora en responderte. Ahora sobre su pregunta:

El Wallet Brick te permite ofrecer pagos desde tu cuenta de Mercado Pago en cualquier etapa del proceso de compra, pero no fue diseñado para acoplarse a un formulario de pago, ejerciendo la funcionalidad de un botón de `submit`. En verdad, los usos recomendables son:

- Con el uso de una [preferencia](https://www.mercadopago.com.ar/developers/es/reference/preferences/_checkout_preferences/post) creada [antes de la inicialización](https://www.mercadopago.com.ar/developers/es/docs/checkout-bricks/wallet-brick/advanced-features/preference-startup) del Brick (caso de uso de ejemplo: _carrito_)
- Sin el uso de una preferencia ya creada, pero que [se crea en el envío](https://www.mercadopago.com.br/developers/es/docs/checkout-bricks/wallet-brick/default-rendering) (ejemplo de caso de uso: _one-click shop_)

Por las características que mencionaste, se recomendaría usar el [Payment Brick](https://www.mercadopago.com.ar/developers/es/live-demo/payment-brick), un formulario completo con varias opciones de pago disponibles, con responsividad, manejo de errores y varias opciones de personalización.

**Reply by icaldana** [reply, marked as answer]
Hola @acyment!

> Capturar el onSubmit y poder frenar el proceso si el form no válida. Cuando hago esto (por ejemplo devolviendo un Promise con reject) el form no hace el submit, pero aparece el cartel de Redirigiendo y el botón no vuelve a aparecer

Sobre ese punto, una posibilidad sería utilizar el flujo de enviar la _preference_ en `submit`, pero solo cuando el formulario se haya completado correctamente. Dejo un ejemplo de cómo se comportaría el Wallet en este caso (con una _preference_ vacía):


https://github.com/mercadopago/sdk-react/assets/30550706/755a7545-5bf5-415f-bd68-6f997bb41461
